packages/management/addons-manager/README.md

```diff
@@ -0,0 +1,43 @@
+# addons-manager Package
+
+Addon-manager generates App CRs for addons based on TKR/BOM, KubeadmControlPlane and CLI inputs provided during cluster creation.
+It then creates these App CRs on management and workload clusters which then get reconciled by a local kapp-controller.
+
+## Components
+
+Client Library
+Addon Controller
+
+## Usage Example
+
+The following is a basic guide for getting started with addons-manager.
+
+Management cluster
+
+1. “tkg init” to create a management cluster.
+
+2. CRS apply raw yaml for kapp-controller. This includes CRDs, Namespaces, RBACs and Deployments for kapp-controller.
+
+3. CRS apply App CR for addon-controller. Kapp-controller will spin up addon-controller.
+
+4. Addon-controller is initialized and watches Cluster(KCP), TanzuKubernetesCluster, TKR/BOM and Secrets with labels tkg.tanzu.vmware.com/addon-name.
+
+5. Addon-controller creates App CRs for kapp-controller. Kapp-controller adopts and manages itself.
+
+6. Addon-controller creates a cluster-admin service account for addons in tkg-system namespace. This cluster-admin service account is just for managed core addons.
+
+7. Addon-controller creates App CRs for all core addons.
+
+8. Kapp-controller reconciles App CRs on the management cluster with a link to the ServiceAccount resource created above and the Secret resource for addon configuration.
+
+Workload cluster
+
+1. “tkg create” to create a workload cluster.
+
+2. Addon-controller creates a cluster-admin service account for addons in tkg-system namespace in the workload cluster. This cluster-admin service account is just for managed core addons.
+
+3. Addon-controller creates App CR in the management cluster for kapp-controller with spec.cluster.kubeconfigSecretRef pointing to the workload cluster. Kapp-controller on management cluster manages kapp-controller on workload cluster.
+
+4. Based on kubeadmControlPlane.spec.Version and TKR, the addon-controller on management cluster will push appropriate App CRs comprising the addons to the workload cluster directly.
+
+5. Kapp-controller in the workload cluster reconciles App CRs to create the addons accordingly.
```

Comment:
suggestion: the info from above steps seems out of date. Like addons-manager no longer creates App CR, it is `packageinstall` now. I suggest we replace those implementation details with a introductory content.

Reply:
I removed most of the outdated description, since we don't have a up to date documentation on addons-controller, I just added very basic way to validate it's installation.